augur/measurements.py

```diff
@@ -0,0 +1,364 @@
+        print(
+            "ERROR: Validation of output JSON failed. See detailed errors above.",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+
+
+def export_measurements(args):
+    # Load input collection TSV file
+    collection_df = load_collection(args['collection'], args['strain_column'], args['value_column'])
+
+    if collection_df is None:
+        print("ERROR: Loading of collection TSV was unsuccessful. See detailed errors above.", file=sys.stderr)
+        sys.exit(1)
+
+    collection_config = {}
+    if args.get('collection_config'):
+        try:
+            collection_config = validate_collection_config_json(args['collection_config'])
+        except ValidateError:
+            print(
+                f"Validation of provided collection config JSON {args['collection_config']} failed. " +
+                "Please check the formatting of this file.",
+                file=sys.stderr
+            )
+            sys.exit(1)
+
+    groupings = collection_config.pop('groupings', None)
+    if args.get('grouping_column'):
+        groupings = get_collection_groupings(collection_df, args['grouping_column'])
+        if collection_config.get('display_defaults', {}).pop('group_by', None):
+            print(
+                "WARNING: The default group-by in the collection config has been removed " +
+                "because new groupings have been provided via the --grouping-column option.",
+                file=sys.stderr
+            )
+
+    if not groupings:
+        print("ERROR: Cannot create measurements JSON without valid groupings", file=sys.stderr)
+        sys.exit(1)
+
+    # Combine collection config with command line args
+    override_config_with_args(collection_config, args)
+
+    # Create collection output object with default values for required keys
+    collection_output = {
+        'key': collection_config.pop('key', os.path.basename(args['collection'])),
+        'groupings': groupings,
+        'x_axis_label': collection_config.pop('x_axis_label', 'measurement values'),
+        'measurements': collection_df.to_dict(orient='records'),
+        **collection_config
+    }
+
+    # Create final output with single collection
+    output = {
+        'collections': [collection_output]
+    }
+
+    # Set indentation to None to create compact JSON if specified
+    indent = {"indent": None} if args['minify_json'] else {}
+    # Create output JSON
+    write_json(output, args['output_json'], include_version=False, **indent)
+    # Verify the produced output is a valid measurements JSON
+    validate_output_json(args['output_json'])
+
+
+def concat_measurements(args):
+    output = {
+        'collections': []
+    }
+    if args.get("default_collection"):
+        output['default_collection'] = args['default_collection']
+
+    for json in args['jsons']:
+        measurements = validate_measurements_json(json)
+        output['collections'].extend(measurements['collections'])
```

Comment:
I would push this to later improvements since it technically doesn't break anything. I can see this as a part of the work to add measurements data as color-by's to the tree.